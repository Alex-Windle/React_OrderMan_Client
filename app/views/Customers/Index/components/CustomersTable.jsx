/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 *
 * @description This redux-form is responsible for displaying information about a given Customer.
 *
 * @description It is also used for creating a Customer.
 *
 */
