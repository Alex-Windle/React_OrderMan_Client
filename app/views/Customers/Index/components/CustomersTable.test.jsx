/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React from 'react';
import renderer from 'react-test-renderer';


// Component imports
import CustomersTable from './CustomersTable';


// Test data imports
import {
  testCustomer,
  testPagination,
} from '../../../../spec/specData';


describe('CustomersTable component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <CustomersTable
        pagination={testPagination}
        handlePageChange={jest.fn()}
        customers={testCustomer}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
