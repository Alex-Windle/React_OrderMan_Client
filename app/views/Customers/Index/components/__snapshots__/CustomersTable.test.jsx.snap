// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersTable component matches snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <table
      className="table table-responsive table-striped table-hover"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Created At
          </th>
          <th
            className="hidden-xs hidden-sm"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="4"
          >
            <h3>
              No Customers Found
            </h3>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="text-center"
  >
    <div>
      <div
        className="container-fluid"
      >
        <ul
          className="pagination"
        >
          <li
            className="disabled"
          >
            <a
              onClick={[Function]}
              role="button"
            >
              «
            </a>
          </li>
          <li
            className="active"
          >
            <a
              onClick={[Function]}
              role="button"
            >
              1
            </a>
          </li>
          <li
            className=""
          >
            <a
              onClick={[Function]}
              role="button"
            >
              2
            </a>
          </li>
          <li
            className=""
          >
            <a
              disabled={false}
              onClick={[Function]}
              role="button"
            >
              »
            </a>
          </li>
        </ul>
      </div>
      <div
        className="container-fluid"
      >
        Showing Results 
        1
         - 
        10
      </div>
    </div>
  </div>
</div>
`;
